ui/src/app/guards/workspace-guard.service.ts

```diff
@@ -3,24 +3,24 @@ import {
   ActivatedRouteSnapshot, CanActivate, CanActivateChild, Router,
   RouterStateSnapshot
 } from '@angular/router';
-import {serverConfigStore} from 'app/utils/navigation';
-import {WorkspacesService} from 'generated';
+import {currentWorkspaceStore, serverConfigStore} from 'app/utils/navigation';
+import {WorkspaceAccessLevel} from 'generated/fetch';
 import {Observable} from 'rxjs/Observable';
 
 @Injectable()
 export class WorkspaceGuard implements CanActivate, CanActivateChild {
   constructor(
-    private workspaceService: WorkspacesService,
     private router: Router) {}
 
   canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
     if (serverConfigStore.getValue().enableResearchReviewPrompt && route.routeConfig.path === 'data' ||
         route.routeConfig.path === 'notebooks') {
-      this.workspaceService.getWorkspace(route.params.ns, route.params.wsid).subscribe(workspace => {
-        if (workspace.accessLevel === 'OWNER'
-            && workspace.workspace.researchPurpose.needsReviewPrompt) {
+      currentWorkspaceStore.subscribe(workspace => {
```

Comment:
Next: this doesn't work during a context switch from workspace A --> workspace B. It would pickup workspace A, whereas you want B here.

I would include here a filter on the workspaceStore to ensure it actually matches the workspace you are guarding for.

e.g. `await currentWorkspaceStore.filter(/* match the ns / ws id */).first().toPromise();`